Library/Homebrew/extend/os/linux/diagnostic.rb

```diff
@@ -0,0 +1,51 @@
+require "tempfile"
+
+module Homebrew
+  module Diagnostic
+    class Checks
+      alias_method :generic_check_tmpdir_sticky_bit, :check_tmpdir_sticky_bit
+
+      def check_tmpdir_sticky_bit
+        message = generic_check_tmpdir_sticky_bit
+        return if message.nil?
+        message + <<~EOS
+          If you don't have administrative privileges on this machine,
+          create a directory and set the HOMEBREW_TEMP environment variable,
+          for example:
+            export HOMEBREW_TEMP=~/tmp
+            mkdir -p "$HOMEBREW_TEMP"
+            chmod 1755 "$HOMEBREW_TEMP"
+            echo "export HOMEBREW_TEMP=\"$HOMEBREW_TEMP\"" >> #{shell_profile}
+        EOS
```

Comment:
should I add a call to `message` after `EOS` ?

Reply:
It's fine like this.